products/apigateway/api.yaml

```diff
@@ -0,0 +1,246 @@
+# Copyright 2020 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Product
+name: ApiGateway
+versions:
+  - !ruby/object:Api::Product::Version
+    name: beta
+    base_url: https://apigateway.googleapis.com/v1beta/
+scopes:
+  - https://www.googleapis.com/auth/cloud-platform
+async: !ruby/object:Api::OpAsync
+  operation: !ruby/object:Api::OpAsync::Operation
+    path: 'name'
+    base_url: '{{op_id}}'
+    wait_ms: 1000
+  result: !ruby/object:Api::OpAsync::Result
+    path: 'response'
+    resource_inside_response: true
+  status: !ruby/object:Api::OpAsync::Status
+    path: 'done'
+    complete: True
+    allowed:
+      - True
+      - False
+  error: !ruby/object:Api::OpAsync::Error
+    path: 'error'
+    message: 'message'
+objects:
+  - !ruby/object:Api::Resource
+    name: 'Api'
+    min_version: beta
+    create_url:  projects/{{project}}/locations/global/apis?apiId={{name}}
+    self_link: projects/{{project}}/locations/global/apis/{{name}}
+    base_url:  projects/{{project}}/locations/global/apis
+    update_verb: :PATCH
+    update_mask: true
+    description: |
+      A consumable API that can be used by multiple Gateways.
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Official Documentation':
+          'https://cloud.google.com/api-gateway/docs/quickstart'
+      api: 'https://cloud.google.com/api-gateway/docs/reference/rest/v1beta/projects.locations.apis'
+    # parameters:
+    #   - !ruby/object:Api::Type::String
+    #     name: 'region'
+    #     description: |
+    #       The name of the Memcache region of the instance.
+    #     required: true
+    #     input: true
+    #     url_param_only: true
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'name'
```

Comment:
name is an output-only field, but apiId is a required URL param. Consider splitting that out into two separate properties.